### PR TITLE
Add Configurable Limits to Controller

### PR DESCRIPTION
**Pull request by dubeejw**
Allows for the configuration of the maximum number of activation records to be displayed for activation list requests, and also allows for the configuration of the blocking action timeout value.

In addition, to making the above values configurable, the mentioned values will also be displayed from the controller's info route.

The goal here is to provide default, min, and max parameter values via controller info route so that the UI/CLI can dynamically read these values.

### CODE REVIEW

**Comment by dubeejw** [comment]
Needs tests.

**Comment by rabbah** [comment]
> Needs tests.
tests usually come first :D

File: ansible/group_vars/all [hunk 1]
```diff
@@ -78,11 +78,15 @@ runtimesManifest:
 defaultLimits:
   actions:
     invokes:
-      perMinute: 120
+      blockingTimeout: 60
```
Comment (rabbah) [inline ansible/group_vars/all]: are you planning to add memory and timeout limits?

**Comment by dubeejw** [comment]
@rabbah, @csantanapr, currently all of the limits/default values are displayed in the controller info route under the limit key. However, I think it would be more organized if the limits were organized in corresponding sections for actions, triggers, packages, etc. See below for examples.

Current format:

```
"limits": {
    "actions_per_minute": 120,
    "action_blocking_timeout": 60,
    "concurrent_actions": 100,
    "activation_poll_record_limit": 200,
    "triggers_per_minute": 60
  }
```

Purposed change:
```
"limits": {
    "actions":
         "invoke": {
              "per_minute": 120,
              "blocking_timeout": 60,
              "concurrent": 100
         }
    },
    "activations": {
         "poll": {
               record_limit": 200
          }
    },
    "triggers": {
         "fire": {
              "per_minute": 60
         }
    }
}
```

**Comment by dubeejw** [comment]
TODO:
Make `MAX_ACTIVATION_ENTITY_LIMIT` configurable via Ansible: /common/scala/src/main/scala/whisk/core/entity/ActivationEntityLimit.scala

File: ansible/environments/local/group_vars/all [hunk 2]
```diff
@@ -17,9 +17,13 @@ db_port: "{{ lookup('ini', 'db_port section=db_creds file={{ playbook_dir }}/db_
...
   actions:
     invokes:
-      perMinute: 60
+      blockingTimeout: 60
       concurrent: 30
       concurrentInSystem: 5000
+      perMinute: 60
+  activations:
+    polls:
+      maxRecords: 200
```
Comment (rabbah) [inline ansible/environments/local/group_vars/all]: i thought there was a limit on all list operations; it looks like for activations it is 200 but for the others, only a default is provided and no limit... this is an oversight.

File: ansible/environments/local/group_vars/all [hunk 3]
```diff
@@ -17,9 +17,13 @@ db_port: "{{ lookup('ini', 'db_port section=db_creds file={{ playbook_dir }}/db_
 limits:
   actions:
     invokes:
-      perMinute: 60
+      blockingTimeout: 60
```
Comment (rabbah) [inline ansible/environments/local/group_vars/all]: maybe this is a better name of this? `requestResponseTimeout`?

File: ansible/environments/mac/group_vars/all [hunk 4]
```diff
@@ -22,9 +22,13 @@ db_port: "{{ lookup('ini', 'db_port section=db_creds file={{ playbook_dir }}/db_
 limits:
   actions:
     invokes:
-      perMinute: 60
+      blockingTimeout: 60
```
Comment (rabbah) [inline ansible/environments/mac/group_vars/all]: since these are the same as the "all" values, i think you can just drop them.

**Comment by rabbah** [comment]
@dubeejw this has rotted - are you still pursuing this?

**Comment by dubeejw** [comment]
@rabbah, still needs to be done. Seems very out of sync with master. Just got side tracked with other work.

**Pull request closed without merging**